### PR TITLE
tools/check: add logging and path sanitization

### LINKED ISSUE #36082: Add logging and path sanitization for tools/check
## Enhancement

The tools located in the "tools/check" directory do not support logging and lack validation of path arguments.

### PR DESCRIPTION
**Pull request by s3nt3**
### What problem does this PR solve?

Issue Number: close #36082 

Problem Summary:

The tools located in the "tools/check" directory do not support logging and lack validation of path arguments.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: tools/check/ut.go [hunk 1]
```diff
@@ -349,11 +351,10 @@ func cmdRun(args ...string) bool {
 }
 
 func parseCaseListFromFile(fileName string) (map[string]struct{}, error) {
-	f, err := os.Open(fileName)
+	f, err := os.Open(filepath.Clean(fileName))
 	if err != nil {
 		return nil, withTrace(err)
 	}
-	//nolint: errcheck
```
Comment (tiancaiamao) [inline tools/check/ut.go:356]: CI success?

File: tools/check/ut.go [hunk 2]
```diff
@@ -810,19 +808,17 @@ func (n *numa) testCommand(pkg string, fn string) *exec.Cmd {
 	}
 	args = append(args, "-test.cpu", "1")
 	if !race {
+		// Don't set timeout for race because it takes a longer when race is enabled.
 		args = append(args, []string{"-test.timeout", "2m"}...)
-	} else {
-		// it takes a longer when race is enabled. so it is set more timeout value.
-		args = append(args, []string{"-test.timeout", "30m"}...)
```
Comment (tiancaiamao) [inline tools/check/ut.go:816]: It seems you're not based on the latest tidb master

File: tools/check/ut.go [hunk 3]
```diff
@@ -810,19 +808,17 @@ func (n *numa) testCommand(pkg string, fn string) *exec.Cmd {
...
 	// session.test -test.run TestClusteredPrefixColum
-	args = append(args, "-test.run", "^"+fn+"$")
+	args = append(args, "-test.run", fn)
+
 	return exec.Command(exe, args...)
 }
 
 func skipDIR(pkg string) bool {
-	skipDir := []string{"br", "cmd", "dumpling", "tests"}
+	skipDir := []string{"br", "cmd", "dumpling"}
```
Comment (tiancaiamao) [inline tools/check/ut.go:821]: ditto

File: Makefile [hunk 4]
```diff
@@ -217,6 +217,10 @@ tools/bin/xprog: tools/check/xprog.go
 	cd tools/check; \
 	$(GO) build -o ../bin/xprog xprog.go
 
+tools/check/go.mod: tools/check/check.mod tools/check/check.sum
```
Comment (xhebox) [inline Makefile:220]: What is the reason of renaming `go.mod` to `check.mod`?

Comment (s3nt3) [reply Makefile:220]: It is a workaround to prevent the SCA tool from scanning linter's third-party dependencies as project dependencies.

**Review by xhebox — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Comment by tiancaiamao** [comment]
image

Poor silly bot...

**Pull request merged**